aws_cloudtrail_rules/aws_kms_cmk_loss.py

```diff
@@ -8,7 +8,9 @@ def rule(event):
 
 
 def dedup(event):
-    for resource in event['resources']:
-        if resource['type'] == KMS_KEY_TYPE:
-            return resource['ARN']
-    return None
+    resources = event.get('resources', {})
+    if resources:
+        for resource in event.get('resources', {}):
+            if resource.get('type', '') == KMS_KEY_TYPE:
+                return resource['ARN']
+    return event.get('p_row_id')
```

Comment:
I'm not sure what exactly you're trying to avoid here, but you could probably re-factor to something like this:

```
for resource in event.get('resources', {}) or {}:
  if resource.get('type', '') == KMS_KEY_TYPE:
    return resource['ARN']
return event.get('p_row_id')
```

At a minimum, you already did `event.get('resources', {})` on line 11, so you can re-use the `resources` variable on line 13.

Reply:
ahh, good point -_-  I'll update it.  (this is to fix warnings/exceptions I saw for when 'resources' doesn't exist)